packages/filesystem/src/common/filesystem.ts

```diff
@@ -17,10 +17,11 @@ export interface FileSystem extends JsonRpcServer<FileSystemClient> {
      * Returns the filestat for the given uri.
      *
      * If the uri points to a folder it will contain one level of unresolved children.
+     * The depth define the number of level for unresolved children. Default value is 0
      *
      * Reject if a file for the given uri does not exist.
      */
-    getFileStat(uri: string): Promise<FileStat>;
+    getFileStat(uri: string, depth?: number | 0): Promise<FileStat>;
```

Comment:
- I would prefer to use the options pattern for optional arguments.
- type number includes type 0, `| 0` is redundant 
- As Sven mentioned 1 is the better default.